### PR TITLE
Adds search UI

### PR DESCRIPTION
**Pull request by fardog**
Adds in a search UI; currently only in the omnibar but it's a component that could be initialized anywhere.

Currently, the search results are a static list stubbed out in `web.clj`

Should be submitting search results over a POST and not a GET, but this requires fixes in how csrf is handled.

2015-12-10-153512_956x1037_scrot

### CODE REVIEW

**Comment by jarofghosts** [comment]
:necktie:

File: src/doctopus/web.clj [hunk 1]
```diff
@@ -96,6 +97,20 @@
...
+  (let [query (:query-params request)
+        {:keys [tentacle-name tentacle-local]} query]
+    (serve-json {:ok true
+                 :tentacle-name tentacle-name
+                 :tentacle-local tentacle-local
+                 :results [{:title "some document"
+                            :snippet "beep boop this is a piece of snippet text"
+                            :url "/some/url"}
+                           {:title "some other document"
+                            :snippet "this is some other piece of snipppet text for yer context"
```
Comment (Gastove) [inline src/doctopus/web.clj]: If `snippet` is missing, will anything break?

File: src/doctopus/web.clj [hunk 2]
```diff
@@ -96,6 +97,20 @@
   [_]
   (serve-html (templates/add-tentacle doctopus)))
 
+(defn serve-search-results
```
Comment (Gastove) [inline src/doctopus/web.clj:111]: Might be time to start moving a lot of our "serve this or that" functions in to their own NS

Comment (Gastove) [inline src/doctopus/web.clj:111]: (That's not at all a todo for you -- thinking out loud)

**Comment by Gastove** [comment]
Diggit :+1:

File: src/doctopus/web.clj (hunk 1 above)
Comment (fardog) [inline src/doctopus/web.clj]: probably! i'll test and make it optional

**Pull request merged**